### PR TITLE
Update for upstream pyth changes

### PR DESCRIPTION
**Pull request by nevi-me**
`get_current_price()` is now `get_price_no_older_than()`, same with EMA.

### CODE REVIEW

**Comment by dnut** [comment]
I'm a little confused by "[DO NOT MERGE]". Am I supposed to look at it? Are you just using it to test your code in CI? Wouldn't making this a draft also make it clear not to merge it?

**Review by zachanon — DISMISSED** [review]

**Review by suscd — COMMENTED** [review]
Why do this everywhere, when the margin program already checks the timestamp on the price?

File: programs/fixed-term/src/margin/instructions/refresh_position.rs [hunk 1]
```diff
@@ -82,15 +84,19 @@ fn load_price(oracle_info: &AccountInfo) -> Result<PositionChange> {
         msg!("oracle error in account {}: {:?}", oracle_info.key, e);
         error!(FixedTermErrorCode::OracleError)
     })?;
+    // Required post pyth-sdk 0.6.1.
+    // See https://github.com/pyth-network/pyth-sdk-rs/commit/4f4f8c79efcee6402a94dd81a0aa1750a1a12297
+    let clock = Clock::get()?;
+    let max_staleness = MAX_ORACLE_STALENESS as u64;
     let price = oracle
-        .get_current_price()
+        .get_price_no_older_than(clock.unix_timestamp, max_staleness)
```
[the remark this answers is not in the document]
Comment (suscd) [reply programs/fixed-term/src/margin/instructions/refresh_position.rs]: looks like the status is checked during conversion to the `PriceFeed` type: https://github.com/pyth-network/pyth-sdk-rs/blob/221a4742f6462986de319299ec0d811356839c7b/pyth-sdk-solana/src/state.rs#L387

File: programs/margin-pool/src/state.rs [hunk 2]
```diff
@@ -343,11 +345,13 @@ impl MarginPool {
     /// Calculate the prices for the deposit and loan notes, based on
     /// the price of the underlying token.
     pub fn calculate_prices(&self, pyth_price: &PriceFeed) -> Result<PriceResult> {
+        let clock = Clock::get()?;
+        let max_staleness = MAX_ORACLE_STALENESS as u64;
         let price_obj = pyth_price
-            .get_current_price()
+            .get_price_no_older_than(clock.unix_timestamp, max_staleness)
```
Comment (suscd) [inline programs/margin-pool/src/state.rs:351]: should be unchecked

File: programs/margin-pool/src/state.rs [hunk 3]
```diff
@@ -343,11 +345,13 @@ impl MarginPool {
...
             .ok_or(ErrorCode::InvalidPoolPrice)?;
         let ema_obj = pyth_price
-            .get_ema_price()
+            .get_ema_price_no_older_than(clock.unix_timestamp, max_staleness)
```
Comment (suscd) [inline programs/margin-pool/src/state.rs:354]: should be unchecked

**Review by suscd — APPROVED** [review]

**Pull request merged**